contracts/halo-pool/src/contract.rs

```diff
@@ -139,67 +196,219 @@ pub fn execute_add_reward_balance(
         res = res.add_submessage(transfer);
     }
 
+    // Update reward balance
+    reward_balance += asset.amount;
+
+    // Save reward balance to phases reward balance
+    phases_reward_balance[phase_index as usize] = reward_balance;
+
     // Update reward_per_second base on new reward balance
     let new_reward_per_second =
-        Decimal::from_ratio(asset.amount, pool_info.end_time - pool_info.start_time).floor();
+        Decimal::from_ratio(reward_balance, pool_info.end_time - pool_info.start_time).floor();
+
     let new_pool_info = PoolInfo {
-        staked_token: pool_info.staked_token.clone(),
-        reward_token: pool_info.reward_token,
         reward_per_second: new_reward_per_second,
-        start_time: pool_info.start_time,
-        end_time: pool_info.end_time,
-        pool_limit_per_user: pool_info.pool_limit_per_user,
-        whitelist: pool_info.whitelist,
+        ..pool_info
     };
 
     // Get staked token balance from pool contract
-    let staked_token_supply =
-        query_token_balance(&deps.querier, pool_info.staked_token, env.contract.address)?;
+    let staked_token_supply = query_token_balance(
```

Comment:
cái này liệu cần không? vì hình như là phase phải start thì mới được stake chứ nhỉ

Reply:
Yes, Removed get staked_token_supply and replaced it to Uint128::zero()